common/persistence/serialization/thrift_mapper.go

```diff
@@ -616,6 +620,46 @@ func transferTaskInfoFromThrift(info *sqlblobs.TransferTaskInfo) *TransferTaskIn
 	}
 }
 
+func crossClusterTaskInfoToThrift(info *CrossClusterTaskInfo) *sqlblobs.CrossClusterTaskInfo {
```

Comment:
We can reuse the sqlblobs.TransferTaskInfo type definition for this and the following function I think?

Reply:
Wouldn't this cause any issues in the future if TransferTaskInfo and CrossClusterTaskInfo diverge in the future? When we serialize we keep the struct name in the blob right? 

We might normally derive both structs from a common ancestor but not sure if Thrift supports inheritance. We can have it as composition but that also looks like a hack?

Reply:
My understanding is that by design/definition transfer task and cross-cluster task are the same task type. The difference is that when the target domain for a transfer task is in a different cluster we put it in the cross cluster queue, call it cross cluster task and process it using a polling method. If target domain did a failover and later become active in the same cluster as the source domain, that cross cluster task will be convert back to be transfer task. Since the processing logic for transfer and cross-cluster task is identical (and we need convert back and forth between them), and taskInfo only contains information needed for processing the task, I think we can reuse the same struct. 

I think we also mentioned this in Database section of the ERD. @yux0 Can you confirm?

I agree that having two definition works and will give us more flexibility. I guess the downside however, is that in the future for any changes we want to make to transfers tasks info, we will need to change both places.

Reply:
Looks like thrift isn't putting the struct name to the serialized data but even then, reusing the same struct name might confuse the developers working on this codebase later. The gain by reusing the same name is less number of lines of code they don't really share the same meaning semantically